ERCS/erc-7439.md

```diff
@@ -0,0 +1,272 @@
+
+* TokenInfo
+    * `signature`: Recommend that the adapter self-defines what to sign using the user's private key or agent's private key to prove the token validity.
+    * `status`: Represent token current status.
+    * `expireTime`: Recommend set to the event due time.
+* TokenStatus
+    * `Sold`: When a token is sold, it MUST change to `Sold`. The token is valid in this status.
+    * `Resell`: When a token is in the secondary market, it MUST be changed to Resell. The token is valid in this status.
+    * `Void`: When the token owner engages in an illegal transaction, the token status MUST be set to Void, and the token is invalid in this status.
+    * `Redeemed`:  When the token is used, it is RECOMMENDED to change the token status to `Redeemed`.
+
+```solidity
+/// @title IERC7439 Prevent Ticket Touting Interface
+interface IERC7439 {
+    /// @dev TokenStatus represent the token current status, only specific role can change status
+    enum TokenStatus {
+        Sold,    // 0
+        Resell,  // 1
+        Void,    // 2
+        Redeemed // 3
+    }
+
+    /// @param signature Data signed by user's private key or agent's private key
+    /// @param tokenStatus Token status changing to
+    /// @param expireTime Event due time
+    struct TokenInfo {
+        bytes signature;
+        TokenStatus tokenStatus;
+        uint256 expireTime;
+    }
+
+    /// @notice Used to notify listeners that the token with the specified ID has been changed status
+    /// @param tokenId The token has been changed status
+    /// @param tokenStatus Token status has been changed to
+    /// @param signature Data signed by user's private key or agent's private key
+    event TokenStatusChanged(
+        uint256 indexed tokenId,
+        TokenStatus indexed tokenStatus,
+        bytes signature
+    );
+
+    /// @notice Used to mint token with token status
+    /// @dev MUST emit the `TokenStatusChanged` event if the token status is changed.
+    /// @param to The receiptent of token
+    /// @param signature Data signed by user's private key or agent's private key
+    function safeMint(address to, bytes memory signature) external;
+
+    /// @notice Used to change token status and can only be invoked by a specific role
+    /// @dev MUST emit the `TokenStatusChanged` event if the token status is changed.
+    /// @param tokenId The token need to change status
+    /// @param signature Data signed by user's private key or agent's private key
+    /// @param tokenStatus Token status changing to
+    /// @param newExpireTime New event due time
+    function changeState(
+        uint256 tokenId,
+        bytes memory signature,
+        TokenStatus tokenStatus,
+        uint256 newExpireTime
+    ) external;
+}
+```
+
+## Rationale
+
+To support customer-oriented resale mechanism and customer-only ticket sale, while also increasing the speculative threshold for speculators to ensure fairness in the market.
```

Comment:
The rationale section should justify technical choices made within the proposal itself. What you have here is justifying the proposal as a whole, and should be in the motivation section.

In this section you could, for example, explain why you've named your mint function `safeMint`, or why tickets must have an expiry time.